client/map_client.go

```diff
@@ -80,3 +80,18 @@ func (c *MapClient) GetAndVerifyMapLeavesByRevision(ctx context.Context, revisio
 	}
 	return c.VerifyMapLeavesResponse(indexes, revision, getResp)
 }
+
+// SetAndVerifyMapLeaves calls SetLeaves and verifies the returned map root.
+func (c *MapClient) SetAndVerifyMapLeaves(ctx context.Context, leaves []*trillian.MapLeaf, metadata []byte) (*types.MapRootV1, error) {
+	// Set new leaf values.
+	req := &trillian.SetMapLeavesRequest{
+		MapId:    c.MapID,
+		Leaves:   leaves,
+		Metadata: metadata,
+	}
+	setResp, err := c.Conn.SetLeaves(ctx, req)
+	if err != nil {
+		return nil, status.Errorf(status.Code(err), "map.SetLeaves(): %v", err)
```

Comment:
Would it be helpful to include the MapID in all of the error messages this client returns?